### DISCUSSION TITLE
Keyboard command to stop/exit a running script

### DISCUSSION THREAD

**Discussion by DaveLPG** [Stellarium/stellarium#3197] [Q&A] [answered]
Try as I might, including RTFM, I could not find how to stop and exit a running Stellarium script.  I did a search for Stop Script in the User Guide - CTRL+D,S (found on page 310) but that did not work.   TIA

**Comment by gzotti** [comment]
The scripts tab in the Config panel, from where you run scripts, has a button to stop them.

**Comment by alex-w** [comment]
> Try as I might, including RTFM, I could not find how to stop and exit a running Stellarium script. I did a search for Stop Script in the User Guide - CTRL+D,S (found on page 310) but that did not work. TIA

It works for me. To stop script press "Ctrl+D", leave keyboard and press "S"

**Comment by DaveLPG** [comment]
Thanks for the quick reply but….

I saw that but when I ran a script, the script took up the entire screen, so how can one get back to the Control panel from a running script?

**Reply by alex-w** [reply]
> Thanks for the quick reply but…. I saw that but when I ran a script, the script took up the entire screen, so how can one get back to the Control panel from a running script? 

Ctrl+T

**Comment by gzotti** [comment]
Press F2 to call the window up again. 
To prevent the window from closing when the script runs, deactivate the tick box "Close window when script runs".

**Comment by DaveLPG** [comment]
OK thanks for that but now I’d like to EXIT the script and go back to where I started from, pre-script. Sorry for the “rookie” questions.

**Comment by gzotti** [comment]
There is no "save state...mess up...go back". The state remains where you stopped the script, which is (almost) like a sequence of user interactions.

**Comment by DaveLPG** [comment]
I did get it to stop (which froze the script on screen) but now I’d like to exit the script and return to the screen from where I started the script.  I should have been more explicit by saying “stop then exit the running script”.  I guess that’s at least a two step operation.

**Comment by DaveLPG** [comment]
So after a stop script one is left with a frozen screen image from which the only way out is CTRL Q?

**Comment by gzotti** [comment]
This must depend on the script. How does the script freeze the screen? And yes, script developing is tedious. JS/Qt experts welcome to improve that.

**Comment by DaveLPG** [comment]
After a CTRL+D , S  the screen freezes (stops) from which apparently, there is no return (exit) other than CTRL Q, unless I’m missing something.

Sorry if this seems to be dragging on, I’m just trying to learn the ways of Stellarium.

**Comment by alex-w** [comment]
пн, 24 апр. 2023 г. в 02:50, DaveLPG ***@***.***>:

> After a CTRL+D , S the screen freezes (stops) from which apparently, there
> is no return (exit) other than CTRL Q, unless I’m missing something.
>
> Sorry if this seems to be dragging on, I’m just trying to learn the ways
> of Stellarium.
>

Ctrl+H ?

-- 
With best regards, Alexander

**Comment by DaveLPG** [comment]
Ctrl+H did some strange things – hard to describe – but it did not exit the running script.

**Reply by gzotti** [reply]
According to the F1/Help panel, Ctrl+H is "Go to home". It should restore home location. Presumably to be pressed after you have stopped the script via Ctrl+D,S (or the stop button, which calls the same internals) and find yourself in a stuck situation. If you press it while the script is running, you mix going to home with your script.

**Comment by DaveLPG** [comment]
After stopping a running script with CTRL+D,S then CTRL+H, I was presented with a circular “home screen” which was not really my home screen but I could see my chosen landscape.  Nothing seemed to work from that home screen so I did a CTRL Q.

**Comment by DaveLPG** [comment]
Here’s what my Home Screen looked like after a CTRL H. Not really my home screen though. Correct landscape but not projection which I have set to Perspective.

![stellarium-007](https://user-images.githubusercontent.com/131614438/234072753-4b0fa5c3-e0ef-4f22-8f9e-ff3a00f3b01a.png)

**Reply by DaveLPG** [reply]
I pasted a screenshot of the result of CTRL-H but it didn't appear above. Not sure why.

**Reply by gzotti** [reply, marked as answer]
Probably because you sent this as email. You can still edit the post. Regardless, there is nothing that can be done here. Script are supposed to run to the end. Cancelling the script may leave the system in whatever state there is.

**Reply by DaveLPG** [reply]
![stellarium-007](https://user-images.githubusercontent.com/131614438/234072506-37861da1-debb-409a-858d-8055c723ca5d.png)
